blog/_posts/2021-10-19-scala-3.1.0-released.md

```diff
@@ -0,0 +1,115 @@
+
+Thanks to Scala 3 strong compatibility guarantees, all code working in 3.0.2 will also work in 3.1. On the other hand, switching to Scala 3.1 allows you to use many improvements and a handful of newly stabilized APIs.
+
+## What's new in 3.1
+
+### New experimental feature: safer exceptions
+
+A new experimental feature that allows declaring and checking which exceptions can be thrown. It relies on the effects as implicit capabilities pattern.
+
+You can opt-in using following import.
+
+```scala
+import language.experimental.saferExceptions
+```
+
+Now it is possible to mark types with exception that can be thrown during the evaluation:
+
+```scala
+def f(x: Double): Double canThrow LimitExceeded =
+  if x < limit then f(x) else throw LimitExceeded()
+```
+
+You can read more in [the document proposing this feature](https://github.com/lampepfl/dotty/blob/release-3.1.0/docs/docs/reference/experimental/canthrow.md).
+
+### Efficient bytecode for matching over strings
+
+Scala, since 3.0.0, has been able to optimize pattern matches with `Int` as a scrutinee type by emitting bytecode containing very efficient instructions: `lookupswitch` or `tableswitch`.
+Now in Scala 3.1, similar behavior was implemented for matching over `String` literals.
+
+```scala
+fruit match
+  case "apple"  => 1
+  case "orange" => 2
+  case "banana" => 3
+  case _        => 0
+```
+
+will now generate `lookupswitch` over hash codes of `String` literals instead of a chain of conditional expressions. This behavior is consistent with what Java does and more performant than the old approach.
+
+### Support for `-Wconf` and `@nowarn`
+
+We have added a `-Wconf` compiler flag that allows filtering and configuring compiler warnings (silence them, or turn them into errors).
+
+We have also brought back `@nowarn` annotation from Scala 2.13, originally inspired by [the silencer plugin](https://github.com/ghik/silencer). It can be used for suppressing warnings directly in the code, in the places where they are expected to occur.
+
+`-Wconf` and `@nowarn` work largely the same way as in Scala 2, but some filters for selecting warnings are different. For the more details see the output of `-Wconf:help` flag.
+
+### Simplified Manifest synthesis
+
+For the sake of compatibility with Scala 2, the Scala 3 compiler now generates given instances for the `Manifest` trait. The new implementation is a slightly simplified approximation of the Scala 2 implementation. It guarantees that any of the expressions:
+
+- `manifest[A] == manifest[B]`
+- `manifest[A].runtimeClass == manifest[B].runtimeClass`
+- `optManifest[A] == optManifest[B]`
+- `optManifest[A].asInstanceOf[ClassTag[A]].runtimeClass == optManifest[B].asInstanceOf[ClassTag[B]].runtimeClass`
+
+that were true in Scala 2, will also remain true when compiled with Scala 3.
+
```

Comment:
I think that https://github.com/lampepfl/dotty/pull/13519 "Raise type error on unreducible match type" is very important to include